app/javascript/article-form/components/ImageUploader.jsx

```diff
@@ -35,136 +36,333 @@ const ImageIcon = () => (
 
 ImageIcon.displayName = 'ImageIcon';
 
-export class ImageUploader extends Component {
-  state = {
+function imageUploaderReducer(state, action) {
+  const { type, payload } = action;
+
+  switch (type) {
+    case 'uploading_image':
+      return {
+        ...state,
+        uploadError: false,
+        uploadingErrorMessage: null,
+        uploadingImage: true,
+        insertionImageUrls: [],
+        showImageCopiedMessage: false,
+      };
+
+    case 'upload_error':
+      return {
+        ...state,
+        insertionImageUrls: [],
+        uploadError: true,
+        uploadErrorMessage: payload.errorMessage,
+        uploadingImage: false,
+      };
+
+    case 'show_copied_image_message':
+      return {
+        ...state,
+        showImageCopiedMessage: true,
+        // showImageCopiedMessage: payload,
+      };
+
+    case 'upload_image_success':
+      return {
+        ...state,
+        uploadingImage: false,
+      };
```

Comment:
The test was failing because the insertion image URLS were not being set.

```suggestion
      return {
        ...state,
        insertionImageUrls: payload.insertionImageUrls,
        uploadingImage: false,
      };
```